code/game/objects/items.dm

```diff
@@ -81,6 +81,8 @@
 	var/list/item_upgrades = list()
 	var/max_upgrades = 3
 
+	var/wallbreaker = FALSE //TIME TO BREAK THE WALL!
```

Comment:
we dont need a variable to check if something is the wallbreaker on the base item class, either do what I suggested handle the code in the hammer or check if(istype(/obj/item/weapon/tool/hammer/ironhammer, I)) in the code for the walls